### PR TITLE
Implementation of `Volta` Integration for Enhanced Development Environment

### LINKED ISSUE #47: Setup volta on the skill-tree-frontend repo
### Issue Description

As part of our ongoing efforts to optimize our development environment and streamline our workflow, we must integrate Volta into our Skill-Tree-Frontend repository. Volta offers a sophisticated solution for managing JavaScript command-line tools, ensuring consistency and efficiency across our projects.
### Expected Behavior

Upon successful integration of Volta, our team members should experience a seamless transition between projects, with Volta automatically managing the versioning of tools like Node, npm, and Yarn based on the project's requirements.
### Current Behavior

At present, our development environment lacks the benefits of Volta integration, leading to potential discrepancies in tool versions across projects and the manual effort required to manage dependencies.
### Reproducibility

 - [x] This issue is reproducible
 - [ ] This issue is not reproducible

### Steps to Reproduce
1.  Navigate to the root directory of the project.
2.  Open the package.json file and you'll see volta is missing

### Severity/Priority
 - [ ] Critical
 - [x] High
 - [ ] Medium
 - [ ] Low

### Additional Information

This integration aligns with our goal of standardizing development practices and enhancing collaboration within our team. By leveraging Volta's capabilities, we anticipate significant improvements in development efficiency and reduced dependency management overhead.
### Checklist

- [x] I have read and followed the project's code of conduct.
- [x] I have searched for similar issues before creating this one.
- [x] I have provided all the necessary information to understand and reproduce the issue.
- [x] I am willing to contribute to the resolution of this issue.

### PR DESCRIPTION
**Pull request by Achintya-Chatterjee**
Date: 1st March 2024

Developer Name: @Achintya-Chatterjee 

---

## Issue Ticket Number

closes #47 #20 
## Description
This PR implements Volta integration into our development environment to streamline the management of JavaScript command-line tools. Volta ensures consistency and compatibility across projects, enhancing our development workflow.

### Documentation Updated?

- [x] Yes
- [ ] No

### Under Feature Flag

- [ ] Yes
- [x] No

### Database Changes

- [ ] Yes
- [x] No

### Breaking Changes

- [ ] Yes
- [x] No

### Development Tested?

- [x] Yes
- [ ] No

## Screenshots
<details>
<summary>Screenshot 1</summary>

[Screencast from 01-03-24 03:02:15 PM IST.webm](https://github.com/Real-Dev-Squad/skill-tree-frontend/assets/55826451/4b2eaca5-f4e6-4710-97a1-e1373887a930)

</details>

## Test Coverage
<details>
<summary>Screenshot 1</summary>

Screenshot from 2024-03-01 14-55-59

</details>

## Additional Notes

- Volta integration ensures consistency in tool management and enhances our development environment. Documentation has been updated accordingly.
why I use the latest version of [yarn](https://yarnpkg.com/migration/overview)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,4 +27,4 @@ yarn
 ```
 yarn dev
 ```
-
+Firstly setup [Volta](https://docs.volta.sh/guide/getting-started) | [Why Volta?](https://docs.volta.sh/guide/#why-volta)
```
Comment (ajoykumardas12) [inline README.md:30]: Need volta before running the server, so this shouldn't be mentioned at last.

Comment (ajoykumardas12) [reply README.md:30]: Yes, check `website-backend` Readme for example

File: README.md [hunk 2]
```diff
@@ -17,6 +17,12 @@ https://stag-skill-tree.vercel.app/
 ### Prerequisites
 - [Node.js](https://nodejs.org/en/download)
 - Yarn 
+- Volta
+
+[Why Volta?](https://docs.volta.sh/guide/#why-volta)
+
+To install Volta, please follow the [process](https://docs.volta.sh/guide/getting-started)
```
Comment (ajoykumardas12) [inline README.md:24]: `... follow this [guide].`

Comment (ajoykumardas12) [reply README.md:24]: Suggestion: Guide would be a better term I think.

File: package.json [hunk 3]
```diff
@@ -58,5 +58,9 @@
         "npm-run-all": "^4.1.5",
         "prettier": "^2.8.8",
```
Comment (iamitprakash) [inline package.json:59]: remove ^ from all places

File: README.md (hunk 2 above)
Comment (Achintya-Chatterjee) [reply README.md:24]: Done. @ajoykumardas12 
Thanks for improving the Developer Experience.

File: package.json (hunk 3 above)
Comment (Achintya-Chatterjee) [reply package.json:59]: Removed.

File: README.md (hunk 2 above)
Comment (ajoykumardas12) [reply README.md:24]: Forgot to push code I guess, don't see changes.

Comment (ajoykumardas12) [reply README.md:24]: I meant this review for line 24

Comment (Achintya-Chatterjee) [reply README.md]: check now.

**Review by ajoykumardas12 — APPROVED** [review]

**Review by bhtibrewal — APPROVED** [review]

**Comment by satyam73** [comment]
Is yarn dev working for you??

it's showing this in mine
image

**Comment by Achintya-Chatterjee** [comment]
> Is yarn dev working for you??
> 
> it's showing this in mine image

@satyam73 it's working,
Screenshot from 2024-03-04 16-26-25

**Pull request merged**